### DISCUSSION TITLE
Some Quesion use BashOperator in version(2.1.4)

### DISCUSSION THREAD

**Discussion by Famer-Y** [apache/airflow#18520] [Q&A] [answered]
command retval 255, but still mark task as success
![image](https://user-images.githubusercontent.com/33817048/134790520-081d316b-a751-4b4a-8035-bf424c61d4b3.png)

**Comment by potiuk** [comment, marked as answer]
This solely depends on what is the bash script you run. The `retval: 255` is likely printed by some of the steps in your script - but not the final one.
Seems you have `piped` commands - in Bash by default the return value of the whole pipe by default is the return value of the last pipe process. So if one of the proceses fails in the pipe, but the last one succeed, the whole pipe result will be 'success' by default.

You can make the whole pipeline fail if any of the pipe commands fail by starting your command with 

`set -o pipefail` 

https://stackoverflow.com/questions/68465355/what-is-the-meaning-of-set-o-pipefail-in-bash-script

So in your case something like that should do:

`set -o pipefail; ps aux | <rest of your command> `

**Reply by Famer-Y** [reply]
Thank You!
